django_filters/fields.py

```diff
@@ -257,16 +269,21 @@ def __init__(self, *args, **kwargs):
 
         super().__init__(*args, **kwargs)
 
-    def _get_choices(self):
-        return super()._get_choices()
-
-    def _set_choices(self, value):
-        super()._set_choices(value)
-        value = self.iterator(self, self._choices)
-
-        self._choices = self.widget.choices = value
-
-    choices = property(_get_choices, _set_choices)
+    @property
+    def choices(self):
+        return super().choices
+
+    @choices.setter
+    def choices(self, value):
+        if DJANGO_50:
+            value = self.iterator(self, value)
+        else:
+            super()._set_choices(value)
+            value = self.iterator(self, self._choices)
```

Comment:
This double setting case is required in Django < 5.0 because callables are handled very specifically whereas it's all handled in `django.utils.choices.normalize_choices()` for Django 5.0.